src/cargo/util/flock.rs

```diff
@@ -373,12 +373,7 @@ mod sys {
     }
 
     pub(super) fn error_unsupported(err: &Error) -> bool {
-        match err.raw_os_error() {
-            Some(libc::ENOTSUP) => true,
-            #[cfg(target_os = "linux")]
-            Some(libc::ENOSYS) => true,
-            _ => false,
-        }
+        matches!(err.raw_os_error(), Some(libc::ENOTSUP) | Some(libc::ENOSYS))
```

Comment:
I believe the meaning here has changed? The `ENOSYS` arm is no longer just on Linux.